server.go

```diff
@@ -152,6 +152,17 @@ type Server struct {
 	// Handler for processing incoming requests.
 	Handler RequestHandler
 
+	// ErrorHandler for returning an error response in user-defined way.
+	//
+	// Following errors will be captured with this handler:
```

Comment:
Since I'm not sure if these are really all the errors (I think more `io.*` errors are possible it's probably better to change this to:
```
The following is a non-exhaustive list of errors that can be expected as argument:
```